File: frameworks/shared/csharp/VelocityBench.HealthCheck/HealthCheck.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace VelocityBench.HealthCheck;

/// <summary>
/// Individual health check result.
/// </summary>
public class HealthCheck
{
    [JsonPropertyName("status")]
    public HealthStatus Status { get; set; }

    [JsonPropertyName("response_time_ms")]
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public double? ResponseTimeMs { get; set; }

    [JsonPropertyName("error")]
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public string? Error { get; set; }

    [JsonPropertyName("warning")]
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public string? Warning { get; set; }

    [JsonPropertyName("info")]
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public string? Info { get; set; }

    [JsonExtensionData]
    public Dictionary<string, object>? AdditionalData { get; set; }

    public HealthCheck(HealthStatus status)
    {
        Status = status;
        AdditionalData = new Dictionary<string, object>();
    }

    public HealthCheck WithResponseTime(double ms)
    {
        ResponseTimeMs = ms;
        return this;
    }

    public HealthCheck WithError(string error)
    {
        Error = error;
        return this;
    }

    public HealthCheck WithWarning(string warning)
    {
        Warning = warning;
        return this;
    }

    public HealthCheck WithInfo(string info)
    {
        Info = info;
        return this;
    }

    public HealthCheck WithData(string key, object value)
    {
        AdditionalData ??= new Dictionary<string, object>();
        AdditionalData[key] = value;
        return this;
    }
}

File: frameworks/shared/php/src/HealthCheck.php
<?php

namespace VelocityBench\HealthCheck;

/**
 * Individual health check result.
 */
class HealthCheck implements \JsonSerializable
{
    public function __construct(
        public HealthStatus $status,
        public ?float $responseTimeMs = null,
        public ?string $error = null,
        public ?string $warning = null,
        public ?string $info = null,
        public array $additionalData = []
    ) {
    }

    public function withResponseTime(float $ms): self
    {
        $this->responseTimeMs = $ms;
        return $this;
    }

    public function withError(string $error): self
    {
        $this->error = $error;
        return $this;
    }

    public function withWarning(string $warning): self
    {
        $this->warning = $warning;
        return $this;
    }

    public function withInfo(string $info): self
    {
        $this->info = $info;
        return $this;
    }

    public function withData(string $key, mixed $value): self
    {
        $this->additionalData[$key] = $value;
        return $this;
    }

    public function jsonSerialize(): array
    {
        $data = ['status' => $this->status->value];

        if ($this->responseTimeMs !== null) {
            $data['response_time_ms'] = $this->responseTimeMs;
        }
        if ($this->error !== null) {
            $data['error'] = $this->error;
        }
        if ($this->warning !== null) {
            $data['warning'] = $this->warning;
        }
        if ($this->info !== null) {
            $data['info'] = $this->info;
        }

        return array_merge($data, $this->additionalData);
    }
}
